stdlib/SuiteSparse_jll/src/SuiteSparse_jll.jl

```diff
@@ -4,137 +4,134 @@
+    const _libcholmod_path = BundledLazyLibraryPath("libcholmod.5.dylib")
+    const _libcolamd_path = BundledLazyLibraryPath("libcolamd.3.dylib")
+    const _libklu_path = BundledLazyLibraryPath("libklu.2.dylib")
+    const _libldl_path = BundledLazyLibraryPath("libldl.3.dylib")
+    const _librbio_path = BundledLazyLibraryPath("librbio.4.dylib")
+    const _libspqr_path = BundledLazyLibraryPath("libspqr.4.dylib")
+    const _libsuitesparseconfig_path = BundledLazyLibraryPath("libsuitesparseconfig.7.dylib")
+    const _libumfpack_path = BundledLazyLibraryPath("libumfpack.6.dylib")
 else
-    const libamd = "libamd.so.3"
-    const libbtf = "libbtf.so.2"
-    const libcamd = "libcamd.so.3"
-    const libccolamd = "libccolamd.so.3"
-    const libcholmod = "libcholmod.so.5"
-    const libcolamd = "libcolamd.so.3"
-    const libklu = "libklu.so.2"
-    const libldl = "libldl.so.3"
-    const librbio = "librbio.so.4"
-    const libspqr = "libspqr.so.4"
-    const libsuitesparseconfig = "libsuitesparseconfig.so.7"
-    const libumfpack = "libumfpack.so.6"
+    const _libamd_path = BundledLazyLibraryPath("libamd.so.3")
+    const _libbtf_path = BundledLazyLibraryPath("libbtf.so.2")
+    const _libcamd_path = BundledLazyLibraryPath("libcamd.so.3")
+    const _libccolamd_path = BundledLazyLibraryPath("libccolamd.so.3")
+    const _libcholmod_path = BundledLazyLibraryPath("libcholmod.so.5")
+    const _libcolamd_path = BundledLazyLibraryPath("libcolamd.so.3")
+    const _libklu_path = BundledLazyLibraryPath("libklu.so.2")
+    const _libldl_path = BundledLazyLibraryPath("libldl.so.3")
+    const _librbio_path = BundledLazyLibraryPath("librbio.so.4")
+    const _libspqr_path = BundledLazyLibraryPath("libspqr.so.4")
+    const _libsuitesparseconfig_path = BundledLazyLibraryPath("libsuitesparseconfig.so.7")
+    const _libumfpack_path = BundledLazyLibraryPath("libumfpack.so.6")
 end
 
-function __init__()
+const libamd = LazyLibrary(_libamd_path, dependencies=_libamd_dependencies)
+const libbtf = LazyLibrary(_libbtf_path, dependencies=_libbtf_dependencies)
+const libcamd = LazyLibrary(_libcamd_path, dependencies=_libcamd_dependencies)
+const libccolamd = LazyLibrary(_libccolamd_path, dependencies=_libccolamd_dependencies)
+const libcholmod = LazyLibrary(_libcholmod_path, dependencies=_libcholmod_dependencies)
+const libcolamd = LazyLibrary(_libcolamd_path, dependencies=_libcolamd_dependencies)
+const libklu = LazyLibrary(_libklu_path, dependencies=_libklu_dependencies)
+const libldl = LazyLibrary(_libldl_path, dependencies=_libldl_dependencies)
+const librbio = LazyLibrary(_librbio_path, dependencies=_librbio_dependencies)
+const libspqr = LazyLibrary(_libspqr_path, dependencies=_libspqr_dependencies)
+const libsuitesparseconfig = LazyLibrary(_libsuitesparseconfig_path, dependencies=_libsuitesparseconfig_dependencies)
+const libumfpack = LazyLibrary(_libumfpack_path, dependencies=_libumfpack_dependencies)
```

Comment:
This is going to be tedious, but you need to express the interdependencies between each of these libraries.  E.g. `libamd` depends on `libsuitesparseconfig`, and `libcholmod` depends on `libamd`, `libcamd`, `libccolamd`, etc...

Reply:
I used a couple of bots to produce and check the list. Just pushed. Does it look reasonable?